source/gameengine/Rasterizer/RAS_Shader.cpp

```diff
@@ -481,6 +481,9 @@ void RAS_Shader::BindAttribute(const char *attr, int loc)
 
 int RAS_Shader::GetUniformLocation(const char *name, bool debug)
 {
+	if (GetError()) {
+		return -1;
+	}
 	BLI_assert(m_shader != NULL);
```

Comment:
Hmm, we have to choose between an assertion or a -1. @youle: What about moving the check in the setTexture function only if the problem appears only in this function.